.travis.yml

```diff
@@ -1,84 +1,69 @@
-language: cpp
-sudo: false
-
-# Use Linux by default
-os: linux
-
-matrix:
-  include:
-    # gcc on arm64
-    - compiler: gcc
-      arch: arm64
-      addons:
-        apt:
-          packages:
-            - ccache
-      env: &Debug
-        - TRAVIS_CUBERITE_BUILD_TYPE=DEBUG
-    # AppleClang
-    # OSX workers are slower to start up.  Having these first in the build matrix makes travis faster overall.
-    - os: osx
-      compiler: clang
-      before_install:
-        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
-      env: &Release
-        - TRAVIS_CUBERITE_BUILD_TYPE=RELEASE
-    - os: osx
-      compiler: clang
-      env: *Debug
-      before_install:
-        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
-    # Default clang
-    - compiler: clang
-      env: *Release
-    - compiler: clang
-      env: *Debug
-    # clang 3.5
-    - compiler: clang
-      dist: trusty
-      addons: &clang35
-        apt:
-          packages:
-            - clang++-3.5
-            - clang-3.5
-      before_install:
-        - CC=clang-3.5;CXX=clang++-3.5
-      env: *Release
-    - compiler: clang
-      dist: trusty
-      addons: *clang35
-      before_install:
-        - CC=clang-3.5;CXX=clang++-3.5
-      env: *Debug
-    # gcc 4.8
-    - compiler: gcc
-      addons: &gcc48
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - g++-4.8
-            - gcc-4.8
-      before_install:
-        - CC=gcc-4.8;CXX=g++-4.8
-      env: *Release
-    - compiler: gcc
-      addons: *gcc48
-      before_install:
-        - CC=gcc-4.8;CXX=g++-4.8
-      env: *Debug
-
-before_script:
-  - export PATH=$(echo "$PATH" | sed -e 's/:\/usr\/lib\/ccache//')
-script: ./travisbuild.sh
-cache: ccache
-
-
-notifications:
-  email:
-    on_success: change
-    on_failure: always
-branches:
-  only:
-    - coverity_scan
-    - master
+language: cpp
+cache: ccache
+
+# Use Linux by default
+os: linux
+dist: bionic
+
+# CMake version out of date: update
+# TODO: add ARM64, PPC, SPARC builds when we find CMake for them
+addons:
+  snaps:
+  - name: cmake
+    confinement: classic
+    channel: latest
+
+matrix:
+  include:
+    # AppleClang Release
+    # OSX workers are slower to start up.  Having these first in the build matrix makes travis faster overall.
+    - os: osx
+      osx_image: xcode11.3
+      before_install:
+        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
+      env: &Release
+        - TRAVIS_CUBERITE_BUILD_TYPE=RELEASE
+
+    # AppleClang Debug
+    - os: osx
+      osx_image: xcode11.3
+      before_install:
+        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
+      env: &Debug
+        - TRAVIS_CUBERITE_BUILD_TYPE=DEBUG
+
+    # Default Clang Release

```

Comment:
We need to pick new minimum compiler versions and test those.

Reply:
g++-8 and clang-7 seems to be the minimum, looking at this table:
https://en.cppreference.com/w/cpp/compiler_support

Reply:
The travis workers seem to be using g++ 7.4 and clang 7.0 which seems alright. We should make sure those versions are pinned in CI and update the README.

Reply:
Travis never upgrades anything, but I can upgrade the readme.